auction-app-backend/src/main/java/com/atlantbh/auctionappbackend/domain/Category.java

```diff
@@ -0,0 +1,154 @@
+package com.atlantbh.auctionappbackend.domain;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonManagedReference;
+
+import javax.persistence.Column;
+import javax.persistence.ColumnResult;
+import javax.persistence.ConstructorResult;
+import javax.persistence.Entity;
+import javax.persistence.ForeignKey;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.NamedNativeQuery;
+import javax.persistence.OneToMany;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.SqlResultSetMapping;
+import javax.persistence.Table;
+import javax.persistence.Transient;
+import java.util.List;
+
+@Entity(name = "Category")
+@Table(name = "category")
+@SqlResultSetMapping(
+        name = "categoryProductCountMapping",
+        classes = @ConstructorResult(
+                targetClass = Category.class,
+                columns = {
+                        @ColumnResult(name = "id", type = Long.class),
+                        @ColumnResult(name = "name", type = String.class),
+                        @ColumnResult(name = "count", type = Long.class)
+                }
+        )
+)
+@NamedNativeQuery(
+        name = "Category.countProductsByCategory",
+        resultClass = Category.class,
+        resultSetMapping = "categoryProductCountMapping",
+        query = "SELECT c.id, c.name, COUNT(c.id) FROM category c, product p WHERE c.id = p.category_id and c.super_category_id = ?1 group by c.id, c.name"
+)
```

Comment:
I figured that it would be more effective this way. If I understand correctly, you are suggesting that when we fetch all categories, then we do something like this `category.getProducts().size()`, of course looping through all fetched categories.